catboost/libs/options/plain_options_helper.cpp

```diff
@@ -389,3 +389,92 @@ void NCatboostOptions::PlainJsonToOptions(
 
     trainOptions["flat_params"] = plainOptions;
 }
+
+void NCatboostOptions::OptionsToPlainJson(
+        const NJson::TJsonValue& options,
+        const NJson::TJsonValue& outputOptions,
+        NJson::TJsonValue* plainOptions)
+{
+    TSet<TString> seenKeys;
```

Comment:
You need to add a check that everything from json is put to seenKeys or explicitly ignored. To do that you can remove seen elements from options and then check that nothing has left.